src/uci.cpp

```diff
@@ -314,8 +314,13 @@ string UCI::value(Value v) {
 
   stringstream ss;
 
-  if (abs(v) < VALUE_MATE_IN_MAX_PLY)
+  if (abs(v) < VALUE_TB_WIN_IN_MAX_PLY)
       ss << "cp " << v * 100 / NormalizeToPawnValue;
+  else if (abs(v) < VALUE_MATE_IN_MAX_PLY)
+  {
+    const int ply = (VALUE_MATE_IN_MAX_PLY - 1) - (std::abs(v) - 1);
+    ss << "cp " << (v > 0 ? 10000 - ply : -10000 + ply);
```

Comment:
Using `VALUE_KNOWN_WIN` rather than `10000` (in both occurrences) would improve the readability.

Reply:
mh idk about that one, we want to keep that value fixed so any changes in the future to `VALUE_KNOWN_WIN`  could but should not be reflected here.. also the meaning of `VALUE_KNOWN_WIN` doesnt really apply here i think

Reply:
+ I think seeing the actual value here is more helpful instead of it being hidden behind an enum... that way you can easily link the line to interested people that just want to have a quick look at it without the need to open an IDE. Ofc you could say this about every value but I've seen more and more people ask about output conversion from stockfish and how its done.

Reply:
10000 is meaningful to give a recognizable value. so, in most cases we will have a 99.XY value for a table base win, with XY being the depth to zero. On entering TB we have a 99.99 value with this choice.

Reply:
> On entering TB we have a 99.99 value with this choice.

Probably a dumb question but would this work fine if the pawn value decreases again? Won't regular evals get past 100?

Reply:
not a dumb question. I think there is no guarantee, but not very likely. The largest an eval can be is 31509 (VALUE_TB_WIN_IN_MAX_PLY - 1) with the current 361 normalization that gets to 87.28. So, guaranteed OK up to a normalization of 316. Now, the largest observed evals are probably way below that bound. The largest eval I would expect is from the classical code on the order of VALUE_KNOWN_WIN (plus a bit) so 10000+ something like 27.70+. I don't know what the largest possible NNUE eval is, but what I have seen as a maximum is quite a bit below that.

My question is if we're happy with the choice of ply, i.e. if we're 'off-by-one' or not. It is not so easy to come up with a test case where this eval can be interpreted :-)

Reply:
I think it would be easier to count if tb was 100 instead of 99.99 right?
My original idea included an uci extension (UCI_ShowTB?) which would make it easier in general. Maybe in the future it can be added.

Reply:
maybe 100 would be easier indeed... uses one more digit of output though :-)

However, admittedly, I'm not quite sure what to expect from this output. On a long line PV, DTZ might go multiple times to 0, just to go up on the next move. So contrary to mate announcement, I'm not sure the expectation that this always decreases is really right.